### ISSUE TITLE
DEVELOPMENT: To have a truly playable game, we now need next:

### ISSUE THREAD

**Issue by makus82** [OpenApoc/OpenApoc#263]
To have a truly playable game, we now need next:

- [ ] Aliens moving from building to building
- [ ] Funding for X-Com and orgs every week
- [ ] Agent Training from battlescape (Agent Training on base implemented) 
- [ ] Bribes for organizations

**Comment by FilmBoy84** [comment]
Don't forget the HUGE unit  behaviours task to be implemented too.
Currently most Agent and Alien actions don't encompass everything they can do in vanilla (crouching, ducking, crawling, using cover, throwing grenades, diversionary assaults, flanking, etc.)

**Comment by FilmBoy84** [comment]
Oh, and Alien Life Cycle properly needs to be implemented so those left-over eggs and chrysalis pose a real threat if not discovered before they hatch in the buildings where they are dropped or spawn.

**Comment by makus82** [comment]
agent actions can be already created but beacose of poor AI its maybe not used 
i dont know....

**Comment by makus82** [comment]
about eggs - its not hatch in current builds?

**Comment by FilmBoy84** [comment]
No eggs/Chrysalis don't seem to hatch after a few days in the cityscape releasing the appropriate alien. This should happen but under 0.1-86 it doesn't seem to.

**Comment by makus82** [comment]
So about agents behavior and actions i think its included here in AI
https://github.com/OpenApoc/OpenApoc/issues/265
And about eggs hatching we need chek it but i add it

**Comment by Atrosha** [comment]
I thought agent training from battlescape already implemented...

**Comment by FilmBoy84** [comment]
@Atrosha Kind of, the calc doesn't seem right, agents get too much XP too quickly additionally androids levelling up REALLY slowly from battle experience (they cant from training) also doesn't appear to work as per OG

**Comment by JonnyH** [comment]
Added:
- [ ] Alien infiltration graph screen

**Comment by makus82** [comment]
some1 already add it)

**Comment by makus82** [comment]
ok "Agent Training from Battlescape " this works check that
but its gives too many skill points at least at superhuman
after 1 mission with 2 agents against 1-2 anthropod 4 spitters 2 hyperworm 3 brainsuckers

so here stats:
agent1 | before | after 
helch | 80 | 80
accur | 36 | 40
react  | 37| 39
speed| 80 | 80
stam | 33 | 41
strng | 58| 64

agent2 | before | after 
helch | 80 | 88
accur | 38 | 40
react  | 16| 17
speed| 80 | 88
stam | 34 | 41
strng | 63| 66

**Comment by makus82** [comment]
About this 
> Alien eggs/Chrysalis don't appear to hatch after some time and spawn new units (Battlescape) - they need to (even if there is rarely a mission long enough for it to coincide with the time they would otherwise spawn in cityscape)
Im really not nothice, playing OG, not even once that eggs/Chrysalis hatch ... but its my expirience ...
So this mechanics dont work? we sure? i will test it ... but maybe some1 already know and we forget check it in list?

**Comment by FilmBoy84** [comment]
It is INCREDIBLY rare, it'll need someone to look at the Original Game code to get the algorithms and values

I also need someone to confirm it's not just my memory failing me as i have a very distinct memory of a UFO mission i did having one spawn a new Anthropod - but i play so much Beta release it could be a memory from those versions (which is why we need to actually check the code and see what is going on)

If, once checked, it turns out to be a pre-final-release feature, we can move to the forums at
http://openapoc.org/forums/new_features/

Because there is now a dedicated area for code changes like this that are not mods (so we don't clog up Data section)

**Comment by Istrebitel** [comment]
Does "Alien eggs/Chrysalis don't appear to hatch after some time and spawn new units (Battlescape) - they need to (even if there is rarely a mission long enough for it to coincide with the time they would otherwise spawn in cityscape)" really belong here?

**Comment by idshibanov** [comment]
> 
> Does **"Alien eggs/Chrysalis don't appear to hatch after some time and spawn new units (Battlescape) - they need to (even if there is rarely a mission long enough for it to coincide with the time they would otherwise spawn in cityscape)"** really belong here?

I agree, this is such a minor thing that shouldn't ever affect somebody's gameplay.

**Comment by makus82** [comment]
We can left it in list but think that it shouldn't affect move to Beta...

**Comment by FilmBoy84** [comment]
I've made a specific road-map for extra or specialist enhancements and features, please see #941
That way we can keep the main three roadmaps clear

Moved the battlescape hatching feature there as Istrebitel has clarified it's not fully implemented in the OG and any instances of it happening I must have from pre-release or messing something up when editing the final release

**Comment by makus82** [comment]
So 8 point what exactly  PR was is that ?
i number them all

**Comment by FilmBoy84** [comment]
@makus82 it's listed as a WIP because it's in PR #998 and we're still reviewing that
Also, issues #999 and #1000 are related (though not essential to leaving alpha - they can be dealt with further down the road)
PR #998 is close to being merged though; not really encountered any issues myself so it's just a final look by other devs and then we should be OK to put it in the master :)

**Comment by makus82** [comment]
and Diplomatic rift event. #996 is part of 7. point? 
so it's all be understandable for new people...

**Comment by makus82** [comment]
how about move 19 point to https://github.com/OpenApoc/OpenApoc/issues/264 ? to close this and after merge all new thing in solid build and then do big post at FB about massive testing of master merged build ...

**Comment by FilmBoy84** [comment]
I'm happy for it to be moved anywhere; but, i feel we are best doing it before any beta release announcement

Have also added point 20 as that definitely needs sorting before we get new users and new bug reports - there needs to be a clear way to identify the version number

I'm for the YYYY-MM-DD-Build Format that many projects have moved to as it's a lot more readable when trying to work out how old a build is given how long we are between major release revisions

But even getting a simple 0.xx.xx system in would be better than 0.0.2-(random string of letters)

**Comment by makus82** [comment]
beta be reached after points be checked in https://github.com/OpenApoc/OpenApoc/issues/264 right?

**Comment by FilmBoy84** [comment]
Yeah, point taken
I'll take issues 19 and 20 off this one, migrate them to #264 just so we can close the Alpha roadmap now it's all being worked on

I'll also un-sticky the Alpha Roadmap and sticky the "post 1.0" #941 after #265 so we are making full use of the "three sticky" limit

**Comment by makus82** [comment]
Btw there weekly funding screen is up n working but there is daily score screen that need to be done...
so need point for this ...

**Comment by FilmBoy84** [comment]
No probs, added the nightly review screen to #264
